basket/news/backends/ctms.py

```diff
@@ -275,13 +275,15 @@ def to_vendor(data, existing_data=None):
         if isinstance(newsletters, dict):
             # Detect unsubscribe all
             optout = data.get("optout", False) or False
-            if (
-                optout
-                and (not any(newsletters.values()))
-                and (set(valid_slugs) == set(newsletters.keys()))
-            ):
+            if optout:
```

Comment:
Does this prevent someone from re-subscribing after they've opted out once? I think it is OK since `optout` would only be sent when unsubscribing, but it is worth integration testing since we don't know what basket sends.

Reply:
I agree we'll need to integration test this. As long as that value isn't repeatedly sent, after a re-subscribe event the newsletters should have the ability to be re-subscribed.

Reply:
Moved this earlier in the new pr.